### PR TITLE
feat: expose all the methods of notification

### PR DESCRIPTION
**Pull request by athakor**

### CODE REVIEW

File: google-cloud-storage/src/main/java/com/google/cloud/storage/NotificationInfo.java [hunk 1]
```diff
@@ -0,0 +1,297 @@
...
+      this.topic = topic;
+      return this;
+    }
+
+    public Builder setPayloadFormat(PayloadFormat payloadFormat) {
+      this.payloadFormat = payloadFormat;
+      return this;
+    }
+
+    /** GcpLaunchStage.Alpha */
```
Comment (rahulKQL) [inline google-cloud-storage/src/main/java/com/google/cloud/storage/NotificationInfo.java]: Please change this to either some meaningful description or remove it.

File: google-cloud-storage/src/main/java/com/google/cloud/storage/NotificationInfo.java [hunk 2]
```diff
@@ -0,0 +1,297 @@
...
+      return this;
+    }
+
+    public NotificationInfo build() {
+      checkNotNull(topic);
+      return new NotificationInfo(this);
+    }
+  }
+
+  NotificationInfo(Builder builder) {
```
Comment (rahulKQL) [inline google-cloud-storage/src/main/java/com/google/cloud/storage/NotificationInfo.java:142]: why can't we have this constructor as a `private`?

File: google-cloud-storage/src/main/java/com/google/cloud/storage/NotificationInfo.java [hunk 3]
```diff
@@ -0,0 +1,297 @@
...
+   *
+   * @see <a href="https://cloud.google.com/storage/docs/cross-origin">Cross-Origin Resource Sharing
+   *     (CORS)</a>
+   */
+  public List<String> getEventTypes() {
+    return eventTypes;
+  }
+
+  /**
+   * Returns the list of additional attributes to attach to each Cloud PubSub message published for\
```
Comment (rahulKQL) [inline google-cloud-storage/src/main/java/com/google/cloud/storage/NotificationInfo.java]: `published for\` => `published for`

File: google-cloud-storage/src/test/java/com/google/cloud/storage/StorageImplTest.java [hunk 4]
```diff
@@ -384,7 +419,8 @@ public long millisTime() {
   private Bucket expectedBucket1, expectedBucket2, expectedBucket3;
 
   @BeforeClass
-  public static void beforeClass() throws NoSuchAlgorithmException, InvalidKeySpecException {
+  public static void beforeClass()
```
Comment (rahulKQL) [inline google-cloud-storage/src/test/java/com/google/cloud/storage/StorageImplTest.java:422]: `beforeClass()` => `setUp()`

File: google-cloud-storage/src/test/java/com/google/cloud/storage/it/ITStorageTest.java [hunk 5]
```diff
@@ -2954,4 +2994,25 @@ public void testBucketLogging() throws ExecutionException, InterruptedException
       RemoteStorageHelper.forceDelete(storage, loggingBucket, 5, TimeUnit.SECONDS);
     }
   }
+
+  @Test
+  public void testGetNotification() {
+    Notification actualNotification = storage.getNotification(BUCKET, notification.getId());
+    System.out.println(actualNotification);
```
Comment (rahulKQL) [inline google-cloud-storage/src/test/java/com/google/cloud/storage/it/ITStorageTest.java]: Please remove this println line

**Review by rahulKQL — CHANGES_REQUESTED** [review]

File: google-cloud-storage/src/test/java/com/google/cloud/storage/StorageImplTest.java (hunk 4 above)
Comment (athakor) [reply google-cloud-storage/src/test/java/com/google/cloud/storage/StorageImplTest.java:422]: it's already there see
https://github.com/googleapis/java-storage/blob/master/google-cloud-storage/src/test/java/com/google/cloud/storage/StorageImplTest.java#L398-L411

File: google-cloud-storage/src/test/java/com/google/cloud/storage/it/ITStorageTest.java (hunk 5 above)
Comment (athakor) [reply google-cloud-storage/src/test/java/com/google/cloud/storage/it/ITStorageTest.java]: done

File: google-cloud-storage/src/main/java/com/google/cloud/storage/NotificationInfo.java (hunk 2 above)
Comment (athakor) [reply google-cloud-storage/src/main/java/com/google/cloud/storage/NotificationInfo.java:142]: done

File: google-cloud-storage/src/main/java/com/google/cloud/storage/NotificationInfo.java (hunk 3 above)
Comment (athakor) [reply google-cloud-storage/src/main/java/com/google/cloud/storage/NotificationInfo.java]: done

File: google-cloud-storage/src/main/java/com/google/cloud/storage/NotificationInfo.java (hunk 1 above)
Comment (athakor) [reply google-cloud-storage/src/main/java/com/google/cloud/storage/NotificationInfo.java]: done

File: google-cloud-storage/src/main/java/com/google/cloud/storage/NotificationInfo.java (hunk 2 above)
Comment (rahulKQL) [reply google-cloud-storage/src/main/java/com/google/cloud/storage/NotificationInfo.java:142]: I still dont see this as a private or any reason for this to be package-private?

**Pull request closed without merging**